lib/kms/file/backend.js

```diff
@@ -0,0 +1,109 @@
+import Common from '../common';
+
+export const backend = {
+    /*
+     * Target implementation will be async. let's mimic it
+     */
+
+
+    /**
+     *
+     * @param {string} bucketName - bucket name
+     * @param {object} log - logger object
+     * @param {function} cb - callback
+     * @returns {undefined}
+     * @callback called with (err, masterKeyId: string)
+    */
+    createBucketKey: function createBucketKeyMem(bucketName, log, cb) {
+        process.nextTick(() => {
+            // Using createDataKey here for purposes of createBucketKeyMem
+            // so that we do not need a separate function.
+            const newKey = Common.createDataKey().toString('hex');
+            cb(null, newKey);
+        });
+    },
+
+
+     /**
+      *
+      * @param {number} cryptoScheme - crypto scheme version number
+      * @param {string} masterKeyId - master key; for the file backend
+      * the master key is the actual bucket master key rather than the key to
+      * retrieve the actual key from a dictionary
+      * @param {buffer} plainTextDataKey - data key
+      * @param {object} log - logger object
+      * @param {function} cb - callback
+      * @returns {undefined}
+      * @callback called with (err, cipheredDataKey: Buffer)
+      */
+    cipherDataKey: function cipherDataKeyMem(cryptoScheme,
```

Comment:
Or well, the callback to that different function is similar, you can pretty much extract it.